marketplace-tests/src/test/java/io/syndesis/qe/MarketplaceTest.java

```diff
@@ -0,0 +1,311 @@
+package io.syndesis.qe;
+
+import static io.syndesis.qe.util.HelperFunctions.copyManifestFilestFromImage;
+import static io.syndesis.qe.util.HelperFunctions.doDeleteRequest;
+import static io.syndesis.qe.util.HelperFunctions.doPostRequest;
+import static io.syndesis.qe.util.HelperFunctions.encodeFileToBase64Binary;
+import static io.syndesis.qe.util.HelperFunctions.getOperatorName;
+import static io.syndesis.qe.util.HelperFunctions.getOperatorVersion;
+import static io.syndesis.qe.util.HelperFunctions.getPackageName;
+import static io.syndesis.qe.util.HelperFunctions.linkPullSecret;
+import static io.syndesis.qe.util.HelperFunctions.readResource;
+import static io.syndesis.qe.util.HelperFunctions.replaceImageInManifest;
+
+import static org.assertj.core.api.Assertions.fail;
+
+import static com.amazonaws.util.StringUtils.UTF8;
+
+import io.syndesis.qe.tar.Compress;
+import io.syndesis.qe.utils.OpenShiftUtils;
+import io.syndesis.qe.utils.TestUtils;
+
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.runner.RunWith;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.lang.RandomStringUtils;
+import org.apache.commons.lang.StringUtils;
+import org.json.JSONObject;
+
+import com.codeborne.selenide.Configuration;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import cucumber.api.CucumberOptions;
+import cucumber.api.junit.Cucumber;
+import io.fabric8.kubernetes.api.model.apps.DeploymentList;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+import lombok.extern.slf4j.Slf4j;
+
+@RunWith(Cucumber.class)
+@Slf4j
+@CucumberOptions(
+    features = "classpath:features",
+    tags = {"not @wip", "not @manual", "not @deprecated", "not @disabled"},
+    plugin = {
+        "pretty",
+        "html:target/cucumber/cucumber-html",
+        "junit:target/cucumber/cucumber-junit.xml",
+        "json:target/cucumber/cucumber-report.json",
+        "io.syndesis.qe.cucumber.MailFormatter:target/cucumber/cucumber-mail/"
+    }
+)
+
+public class MarketplaceTest {
+
+    private static String quayPackage;
+
+    private static final String QUAY_LOGIN_URL = "https://quay.io/cnr/api/v1/users/login";
+    private static final String QUAY_PUSH_URL = "https://quay.io/cnr/api/v1/packages/" + TestConfiguration.quayNamespace();
+    private static final String QUAY_REPOSITORY_URL =  "https://quay.io/api/v1/repository/" +
+        TestConfiguration.quayNamespace() + "/" + "QUAY_PACKAGE";
+    private static final String QUAY_CHANGE_VISIBILITY_URL = QUAY_REPOSITORY_URL + "/changevisibility";
+
+    @BeforeClass
+    public static void setupMarketplace() throws IOException {
+        quayPackage = createQuayProject();
+
+        deployOperatorToOpenshhift(quayPackage);
```

Comment:
can the "setup" and "teardown" be implemented as `Resource` instance in https://github.com/syndesisio/syndesis-qe/tree/master/utilities/src/main/java/io/syndesis/qe/resource/impl ?